### PR TITLE
Use dama/doctrine-test-bundle to improve phpunit tests

### PR DESCRIPTION
**Pull request by dmaicher**
As also done in https://github.com/Sylius/Sylius/pull/7497/files

### CODE REVIEW

File: phpunit.xml.dist [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+         bootstrap="app/autoload.php">
+    <testsuites>
+        <testsuite name="Sylius Standard Test Suite">
+            <directory>tests</directory>
+        </testsuite>
+    </testsuites>
+
+    <php>
+        <server name="KERNEL_CLASS_PATH" value="/app/TestAppKernel.php" />
+        <server name="IS_DOCTRINE_ORM_SUPPORTED" value="true" />
```
Comment (dmaicher) [inline phpunit.xml.dist]: this can be removed maybe?

File: app/config/config_test.yml [hunk 2]
```diff
@@ -15,7 +15,6 @@ framework:
 doctrine:
     dbal:
         dbname: "%database_name%_test"
-        driver_class: Sylius\Bundle\CoreBundle\Test\MySqlDriver
```
Comment (dmaicher) [inline app/config/config_test.yml]: seems this is a different class than the one I deleted here: https://github.com/Sylius/Sylius/pull/7497/files#diff-89e907faa9c86340f8a0ae7df6ecdfaaL12

Comment (pamil) [inline app/config/config_test.yml]: Sylius-Standard was upgraded before that class was renamed so it's all right :+1:

File: phpunit.xml.dist (hunk 1 above)
Comment (pamil) [inline phpunit.xml.dist]: It's specific to Lakion/ApiTestCase and I guess it is still required. /cc @lchrusciel

File: composer.json [hunk 3]
```diff
@@ -45,7 +45,8 @@
         "phpunit/phpunit": "^5.6",
         "se/selenium-server-standalone": "^2.52",
         "sensio/generator-bundle": "^3.1",
-        "stripe/stripe-php": "^4.1"
+        "stripe/stripe-php": "^4.1",
+        "dama/doctrine-test-bundle": "^1.0"
```
Comment (pamil) [inline composer.json]: It would be awesome to keep alphabetical order here 🎉

File: phpunit.xml.dist (hunk 1 above)
Comment (lchrusciel) [inline phpunit.xml.dist]: It has to stay, because otherwise fixtures will not be loaded.

File: composer.json (hunk 3 above)
Comment (dmaicher) [inline composer.json]: yeah just used `composer require --dev dama/doctrine-test-bundle` :yum: 

I can restore the order again :wink:

**Comment by dmaicher** [comment]
@pamil I can also remove the `phpunit.xml.dist` again. I just thought it might be a nice start to have a basic config already.

**Comment by dmaicher** [comment]
Changes done.

I also updated `sylius/sylius` to latest master commit.

**Comment by pjedrzejewski** [comment]
Great, thank you so much David! :)

**Pull request merged**